src/Domain/Services/CheckoutFields.php

```diff
@@ -0,0 +1,644 @@
+				'index'          => 30,
+			),
+			'address_1'  => array(
+				'label'          => __( 'Address', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'Address (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => true,
+				'hidden'         => false,
+				'autocomplete'   => 'address-line1',
+				'autocapitalize' => 'sentences',
+				'index'          => 40,
+			),
+			'address_2'  => array(
+				'label'          => __( 'Apartment, suite, etc.', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'Apartment, suite, etc. (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => false,
+				'hidden'         => false,
+				'autocomplete'   => 'address-line2',
+				'autocapitalize' => 'sentences',
+				'index'          => 50,
+			),
+			'country'    => array(
+				'label'         => __( 'Country/Region', 'woo-gutenberg-products-block' ),
+				'optionalLabel' => __(
+					'Country/Region (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'      => true,
+				'hidden'        => false,
+				'autocomplete'  => 'country',
+				'index'         => 50,
+			),
+			'city'       => array(
+				'label'          => __( 'City', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'City (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => true,
+				'hidden'         => false,
+				'autocomplete'   => 'address-level2',
+				'autocapitalize' => 'sentences',
+				'index'          => 70,
+			),
+			'state'      => array(
+				'label'          => __( 'State/County', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'State/County (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => true,
+				'hidden'         => false,
+				'autocomplete'   => 'address-level1',
+				'autocapitalize' => 'sentences',
+				'index'          => 80,
+			),
+			'postcode'   => array(
+				'label'          => __( 'Postal code', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'Postal code (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => true,
+				'hidden'         => false,
+				'autocomplete'   => 'postal-code',
+				'autocapitalize' => 'characters',
+				'index'          => 90,
+			),
+			'phone'      => array(
+				'label'          => __( 'Phone', 'woo-gutenberg-products-block' ),
+				'optionalLabel'  => __(
+					'Phone (optional)',
+					'woo-gutenberg-products-block'
+				),
+				'required'       => false,
+				'hidden'         => false,
+				'type'           => 'tel',
+				'autocomplete'   => 'tel',
+				'autocapitalize' => 'characters',
+				'index'          => 100,
+			),
+		);
+
+		$this->fields_locations = array(
+			// omit email from shipping and billing fields.
+			'address'    => array_merge( \array_diff_key( array_keys( $this->core_fields ), array( 'email' ) ) ), // everything here will be saved to customer and order.
```

Comment:
That's a good but something to figure out later because we don't allow reordering fields yet or placing them in specific places. Currently, email is the only one not in the address, but was previously handled manually and part of the indexing.